src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/BookBinder.java

```diff
@@ -0,0 +1,121 @@
+package io.github.thebusybiscuit.slimefun4.implementation.items.electric.machines;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.annotation.ParametersAreNonnullByDefault;
+
+import org.bukkit.Material;
+import org.bukkit.enchantments.Enchantment;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.meta.EnchantmentStorageMeta;
+
+import io.github.thebusybiscuit.cscorelib2.inventory.InvUtils;
+import io.github.thebusybiscuit.slimefun4.api.items.ItemSetting;
+import me.mrCookieSlime.Slimefun.Lists.RecipeType;
+import me.mrCookieSlime.Slimefun.Objects.Category;
+import me.mrCookieSlime.Slimefun.Objects.SlimefunItem.abstractItems.AContainer;
+import me.mrCookieSlime.Slimefun.Objects.SlimefunItem.abstractItems.MachineRecipe;
+import me.mrCookieSlime.Slimefun.api.SlimefunItemStack;
+import me.mrCookieSlime.Slimefun.api.inventory.BlockMenu;
+
+public class BookBinder extends AContainer {
+
+    private final ItemSetting<Boolean> deniesVanillaRules = new ItemSetting<>("denies-vanilla-rules", true); 
+
+    @ParametersAreNonnullByDefault
+    public BookBinder(Category category, SlimefunItemStack item, RecipeType recipeType, ItemStack[] recipe) {
+        super(category, item, recipeType, recipe);
+        
+       addItemSetting(deniesVanillaRules); 
+    }
+
+    @Override
+    protected MachineRecipe findNextRecipe(BlockMenu menu) {
+        for (int slot : getInputSlots()) {
+            ItemStack target = menu.getItemInSlot(slot == getInputSlots()[0] ? getInputSlots()[1] : getInputSlots()[0]);
+
+            if (!isCompatible(target)) {
+                return null;
+            }
+
+            ItemStack item = menu.getItemInSlot(slot);
+            if  (item != null && item.getType() == Material.ENCHANTED_BOOK && target != null) {
+                Map<Enchantment, Integer> enchantments = new HashMap<>();
+                int amount = 0;
+                EnchantmentStorageMeta itemMeta = (EnchantmentStorageMeta) item.getItemMeta();
+                EnchantmentStorageMeta targetMeta = (EnchantmentStorageMeta) target.getItemMeta();
+
+                for(Map.Entry<Enchantment, Integer> e : itemMeta.getStoredEnchants().entrySet()) {
+                    for (Map.Entry<Enchantment, Integer> e2 : targetMeta.getStoredEnchants().entrySet()) {
+                        if (e.getKey() == e2.getKey()) {
+                            if (e.getValue() == e2.getValue()) {
+                                enchantments.put(e.getKey(), e.getValue()+1);
+                            } else {
+                                enchantments.put(e.getKey(), e.getValue() > e2.getValue() ? e.getValue() : e2.getValue());
+                            }
+                        }
+                        enchantments.put(e2.getKey(), e2.getValue());
+                    }
+
+                    amount++;
+                    enchantments.put(e.getKey(), e.getValue());
+                }
```

Comment:
This section could be replaced with a more elegant solution.

```suggestion
                Map<Enchantment, Integer> storedEnchantments = itemMeta.getStoredEnchants();
                enchantments.putAll(storedEnchantments);
                for (Map.Entry<Enchantment, Integer> entry : storedEnchantments.entrySet()) {
                    enchantments.merge(entry.getKey(), entry.getValue(), (a, b) -> {
                            if (a == b) {
                                return a + 1;
                            } else {
                                return Math.max(a, b);
                            }
                    });
                }
```

I haven't tested this, so I might be missing a semicolon or similar somewhere. But i think this would be much cleaner.